presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkTaskExecution.java

```diff
@@ -339,25 +349,26 @@ private DriverSplitRunnerFactory(DriverFactory driverFactory, boolean partitione
             this.pipelineContext = taskContext.addPipelineContext(driverFactory.getPipelineId(), driverFactory.isInputDriver(), driverFactory.isOutputDriver(), partitioned);
         }
 
-        public DriverSplitRunner createDriverRunner(@Nullable ScheduledSplit partitionedSplit)
+        public DriverSplitRunner createDriverRunner(@Nullable List<ScheduledSplit> scheduledSplits)
         {
             checkState(!noMoreDriverRunner.get(), "Cannot create driver for pipeline: %s", pipelineContext.getPipelineId());
             pendingCreation.incrementAndGet();
             // create driver context immediately so the driver existence is recorded in the stats
-            // the number of drivers is used to balance work across nodes
-            long splitWeight = partitionedSplit == null ? 0 : partitionedSplit.getSplit().getSplitWeight().getRawValue();
-            DriverContext driverContext = pipelineContext.addDriverContext(splitWeight, Lifespan.taskWide(), driverFactory.getFragmentResultCacheContext());
-            return new DriverSplitRunner(this, driverContext, partitionedSplit);
+            // splitWeight can be 0 as we don't load balance the executor based on their load average
+            DriverContext driverContext = pipelineContext.addDriverContext(0, Lifespan.taskWide(), driverFactory.getFragmentResultCacheContext());
+            DriverSplitRunner r = new DriverSplitRunner(this, driverContext, scheduledSplits);
+            r.getInfo();
+            return r;
         }
 
-        public Driver createDriver(DriverContext driverContext, @Nullable ScheduledSplit partitionedSplit)
+        public Driver createDriver(DriverContext driverContext, @Nullable List<ScheduledSplit> scheduledSplits)
         {
             Driver driver = driverFactory.createDriver(driverContext);
-            if (partitionedSplit != null) {
+            if (scheduledSplits != null && scheduledSplits.size() > 0) {
                 boolean isNativeExecutionEnabled = isNativeExecutionEnabled(driver.getDriverContext().getSession());
-                PlanNodeId sourceNodeId = isNativeExecutionEnabled ? driver.getSourceId().get() : partitionedSplit.getPlanNodeId();
+                PlanNodeId sourceNodeId = isNativeExecutionEnabled ? driver.getSourceId().get() : Iterables.getOnlyElement(scheduledSplits).getPlanNodeId();
```

Comment:
`Iterables.getOnlyElement` will throw exception if the list contains multiple element, should we do a pre-check?

Reply:
If it is not the only element in a non-native execution, we likely are doing something really wrong exampke: not evaluating all splits that were assigned.
So I think the exception is needed.

Reply:
Yeah, I mean we should explicitly check the list only containing one element and throw a more meaningful exception msg if not (say `native execution requires only one scheduledSplits but [%d] were found.`) to provide more debugging context.